### PR TITLE
SimpleCov Support

### PR DESCRIPTION
**Pull request by etagwerker**
Hey @Onumis @TreyE and friends! 

This PR is an attempt to solve a part of https://github.com/whitesmith/rubycritic/issues/245

For now it just lists files and its coverage in a new section. It also addresses an item in the [ROADMAP.md](ROADMAP.md): 

>  Integrate other gems, like:
> 
>  Simplecov to provide code coverage

Before I work more on this, I figured I could send the PR and get your feedback. A couple of questions that I had: 

1. Do you think it is a good idea to add it to RubyCritic's core? 
2. Or, do you think it should be a separate gem? (e.g. rubycritic-simplecov) 

A lot of the code that is in `lib/rubycritic/analysers/coverage.rb` comes from SimpleCov's merger: https://github.com/colszowka/simplecov/blob/master/lib/simplecov/result_merger.rb#L14-L115 (sorry I didn't find a classier way) 

Here is what it looks like: 

ruby-critic-coverage

Compared to SimpleCov's report: 

simplecov-coverage-gggiiifff-com

This is the sample application that I used: https://github.com/fastruby/gggiiifff.com

Please check it out and let me know if I am on the right track. 😄 

Thanks!

### CODE REVIEW

**Comment by etagwerker** [comment]
@Onumis @TreyE just fixed the failing specs. Let me know what you think.

File: lib/rubycritic/analysers/coverage.rb [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      end
+
+      # Loads the cached resultset from JSON and returns it as a Hash,
+      # caching it for subsequent accesses.
+      def resultset
+        @resultset ||= begin
+          if (data = stored_data)
+            begin
+              JSON.parse(data) || {}
+            rescue
```
Comment (Onumis) [inline lib/rubycritic/analysers/coverage.rb]: This should rescue from `JSON::ParserError` or at least `StandardError` and probably print to STDOUT.
It should not happen, but if it does the user would like to know.

**Comment by etagwerker** [comment]
@Onumis Just addressed the issues you mentioned. Please let me know if it is good now.

**Comment by etagwerker** [comment]
@Onumis Did you see anything else that needs to be improved? Just curious... 😸

**Comment by Onumis** [comment]
LGTM! 🎉 
Will merge it now and prepare a release later today. 
Thanks! 🙏

**Comment by silvermind** [comment]
Hi @etagwerker nice PR 👍 
As we're using special coverage dir in the `rails_helper.rb` for the Specs (coverage_dir ENV.fetch('SIMPLE_COV_OUTPUT_DIR')), we don't get any results in rubycritc context.
As this config is only loaded in rspec context, moving the same to the `development.rb` or `test.rb` environments didn't help too, as rubycritic does not load them at all.. any ideas how to solve this in the rubycritic context?

**Comment by etagwerker** [comment]
@silvermind I'm not sure what could be wrong there. I wrote the PR so that it uses `SimpleCov.coverage_dir` when trying to find the `.resultset.json` file. Could you open an issue here? https://github.com/whitesmith/rubycritic/issues/new

That way we can continue the conversation over there.

**Pull request merged**